src/fprime_gds/common/loaders/json_loader.py

```diff
@@ -0,0 +1,218 @@
+import fprime.common.models.serialize.numerical_types as numerical_types
+from fprime.common.models.serialize.serializable_type import SerializableType
+from fprime.common.models.serialize.string_type import StringType
+from fprime.common.models.serialize.type_base import BaseType
+
+from fprime_gds.common.utils.string_util import preprocess_fpp_format_str
+from fprime_gds.common.loaders import dict_loader
+
+
+PRIMITIVE_TYPE_MAP = {
+    "I8": numerical_types.I8Type,
+    "I16": numerical_types.I16Type,
+    "I32": numerical_types.I32Type,
+    "I64": numerical_types.I64Type,
+    "U8": numerical_types.U8Type,
+    "U16": numerical_types.U16Type,
+    "U32": numerical_types.U32Type,
+    "U64": numerical_types.U64Type,
+    "F32": numerical_types.F32Type,
+    "F64": numerical_types.F64Type,
+    "bool": BoolType,
+}
+
+
+class JsonLoader(dict_loader.DictLoader):
+    """Class to help load JSON dictionaries"""
+
+    # Cache parsed type objects at the class level so they can be reused across subclasses
+    parsed_types: dict = {}
+
+    def __init__(self, json_file: str):
+        """
+        Constructor
+
+        Returns:
+            An initialized loader object
+        """
+        super().__init__()
+        with open(json_file, "r") as f:
+            self.json_dict = json.load(f)
+
+    def get_versions(self):
+        """
+        Get the framework and project versions of the dictionary
+
+        Returns:
+            A tuple of the framework and project versions
+        """
+        if "metadata" not in self.json_dict:
+            # REVIEW NOTE: Should we raise an error instead?
+            return ("unknown", "unknown")
+        return (
+            self.json_dict.get("metadata").get("frameworkVersion", "unknown"),
+            self.json_dict.get("metadata").get("projectVersion", "unknown"),
+        )
+
+    def parse_type(self, type_dict: dict) -> BaseType:
+        type_name: str = type_dict.get("name", None)
+
+        if type_name is None:
+            raise ValueError(
+                f"Channel entry in dictionary has no `name` field: {str(type_dict)}"
+            )
+
+        if type_name in PRIMITIVE_TYPE_MAP:
+            return PRIMITIVE_TYPE_MAP[type_name]
+
+        if type_name == "string":
+            # REVIEW NOTE: All strings of same size will share the same type. I believe this is good?
+            return StringType.construct_type(
+                f'String_{type_dict.get("size")}', type_dict.get("size")
+            )
+
+        # Check if type has already been parsed
+        if type_name in self.parsed_types:
+            return self.parsed_types[type_name]
+
+        # Parse new enum/array/serializable types
+        qualified_type = None
+        for type_def in self.json_dict.get("typeDefinitions", []):
+            if type_name == type_def.get("qualifiedName"):
+                qualified_type = type_def
+                break
+
+        if qualified_type is None:
```

Comment:
Done. I don't use for-else much because it's very python-specific and not always intuitive what it's doing. Especially here where it could look like an indentation error.